pkg/mutation/schema/node_test.go

```diff
@@ -154,6 +167,58 @@ func TestNode_Add(t *testing.T) {
 				id("list-object"):   true,
 			},
 		},
+		{
+			name: "don't need to terminate",
+			before: []idPath{
+				ipmt("object", "spec.fields.foo", false),
+			},
+			add:  ip("more fields", "spec.fields.foo.bar"),
+			want: nil,
+		},
+		{
+			name: "must terminate for before",
+			before: []idPath{
+				ipmt("object", "spec.fields.foo", true),
+			},
+			add: ip("more fields", "spec.fields.foo.bar"),
+			want: IDSet{
+				id("object"):      true,
+				id("more fields"): true,
+			},
+		},
+		{
+			name: "must terminate for after",
+			before: []idPath{
+				ip("more fields", "spec.fields.foo.bar"),
+				ip("same-path", "spec.fields.foo"),
+			},
+			add: ipmt("object", "spec.fields.foo", true),
+			want: IDSet{
+				id("object"):      true,
+				id("more fields"): true,
+			},
+		},
+		{
+			name: "must terminate for before and after",
+			before: []idPath{
+				ipmt("object", "spec.fields.foo", true),
+				ip("object-2", "spec.fields.foo"),
+			},
+			add: ipmt("fields", "spec.fields", true),
+			want: IDSet{
+				id("object"):   true,
+				id("object-2"): true,
+				id("fields"):   true,
+			},
+		},
+		{
+			name: "must terminate for before with no conflict",
+			before: []idPath{
+				ipmt("object", "spec.fields.foo", true),
+			},
+			add:  ip("object-2", "spec.fields.foo"),
```

Comment:
I think (under the current version of the code) if you added another test where line 219 went from:

`add:  ip("object-2", "spec.fields.foo"),`

to:

`add:  ip("object-2", "spec.fields.baz"),`

you'd see a conflict reported when there should be none.